### DISCUSSION TITLE
case.build error: (.text+0x24): undefined reference to `main'

### DISCUSSION THREAD

**Discussion by J-shel** [E3SM-Project/E3SM#5621] [E3SM model help]
My environments:
Nvidia HPC SDK 20.7
mpirun (Open MPI) 3.1.5

Hello,

I met an error when building compset WCYCL1850 using Nvidia compiler and OpenMPI。

The following steps go well:
$./create_newcase --case WCYCL1850_11 --compset WCYCL1850 --res ne30pg2_EC30to60E2r2 --mach saber --compiler nvidia --mpilib openmpi
$./case.setup

And the error occured during ./case.build:

[100%] Linking CXX executable /e3sm/e3sm_2_1/output/scratch/WCYCL1850_11/bld/e3sm.exe
cd /e3sm/e3sm_2_1/output/scratch/WCYCL1850_11/bld/cmake-bld/cmake/cpl && /usr/bin/cmake -E cmake_link_script CMakeFiles/e3sm.exe.dir/link.txt --verbose=1
python3 /e3sm/e3sm_2_1/E3SM-maint2.1-git_all_nvhpcsdk/cime/scripts/WCYCL1850_11/Tools/e3sm_compile_wrap.py  
......
/usr/lib64/crt1.o: In function `_start':
(.text+0x24): undefined reference to `main'
Target /e3sm/e3sm_2_1/output/scratch/WCYCL1850_11/bld/e3sm.exe built in 1.041719 seconds
gmake[2]: *** [cmake/cpl/CMakeFiles/e3sm.exe.dir/build.make:555: /e3sm/e3sm_2_1/output/scratch/WCYCL1850_11/bld/e3sm.exe] Error 2
gmake[2]: Leaving directory '/e3sm/e3sm_2_1/output/scratch/WCYCL1850_11/bld/cmake-bld'
gmake[1]: *** [CMakeFiles/Makefile2:646: cmake/cpl/CMakeFiles/e3sm.exe.dir/all] Error 2
gmake[1]: Leaving directory '/e3sm/e3sm_2_1/output/scratch/WCYCL1850_11/bld/cmake-bld'
gmake: *** [Makefile:94: all] Error 2
real 448.72
user 2313.72
sys 234.72

In addition, I using nvidia.cmake file. 
Thanks!

**Comment by J-shel** [comment]
I'm using E3SM v2.1.

**Comment by rljacob** [comment]
Have you compiled a simple mpi program with this system?

It looks like you may have the C or C++ linker set in your compiler config instead of Fortran.  Try the X case which is for testing a new machine:  --res f19_g16 --compset X
